test/OpenTelemetry.Tests/Metrics/MetricAPITest.cs

```diff
@@ -382,6 +382,168 @@ public void ObservableCounterAggregationTest(bool exportDelta)
             }
         }
 
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void DimensionsAreOrderInsensitiveWithSortedKeysFirst(bool exportDelta)
+        {
+            var exportedItems = new List<Metric>();
+
+            using var meter = new Meter($"{Utils.GetCurrentMethodName()}.{exportDelta}");
+            var counterLong = meter.CreateCounter<long>("Counter");
+            using var meterProvider = Sdk.CreateMeterProviderBuilder()
+                .AddMeter(meter.Name)
+                .AddReader(new BaseExportingMetricReader(new InMemoryExporter<Metric>(exportedItems))
+                {
+                    Temporality = exportDelta ? AggregationTemporality.Delta : AggregationTemporality.Cumulative,
+                })
+                .Build();
+
+            // Emit the first metric with the sorted order of tag keys
+            counterLong.Add(5, new("Key1", "Value1"), new("Key2", "Value2"), new("Key3", "Value3"));
+            counterLong.Add(10, new("Key1", "Value1"), new("Key3", "Value3"), new("Key2", "Value2"));
+            counterLong.Add(10, new("Key2", "Value20"), new("Key1", "Value10"), new("Key3", "Value30"));
+
+            // Emit a metric with different set of keys but the same set of values as one of the previous metric points
+            counterLong.Add(25, new("Key4", "Value1"), new("Key5", "Value3"), new("Key6", "Value2"));
+
+            meterProvider.ForceFlush(MaxTimeToAllowForFlush);
+
+            List<KeyValuePair<string, object>> tagsForFirstMetricPoint = new List<KeyValuePair<string, object>>()
```

Comment:
nit: rename to expectedTags..

Reply:
anyway to use the same while doing the measurement as well? counter.Add(10, tags1).

Reply:
We can use `TagList` but then we will end up creating quite a few `TagList`s one for each combination of the dimensions.